MCprep_addon/spawner/models.py

```diff
@@ -0,0 +1,1350 @@
+					util.select_set(ob, False)
+			try:
+				importedObj = bpy.context.selected_objects[0]
+			except:
+				print("selected object not found")
+				self.report({'WARNING'}, "Imported object not found")
+				return {'CANCELLED'}
+			importedObj["MCprep_noSwap"] = 1
+			importedObj.location = self.location
+			if self.prep_materials is True:
+				try:
+					bpy.ops.mcprep.prep_materials(
+						autoFindMissingTextures=False,
+						improveUiSettings=False,
+						skipUsage=True) # if cycles
+				except:
+					print("Could not prep materials")
+					self.report({"WARNING"}, "Could not prep materials")
+		else:
+			if not use_cache:
+				group = self.prep_collection(context, block, pre_groups)
+
+			if not group:
+				conf.log("No group identified, could not retrieve imported group")
+				self.report({"ERROR"}, "Could not retrieve imported group")
+				return {'CANCELLED'}
+
+			# now finally add the group instance
+			for obj in util.get_objects_conext(context):
+				util.select_set(obj, False)
+			ob = util.addGroupInstance(group.name, self.location)
+			if self.snapping=="center":
+				offset=0 # could be 0.5
+				ob.location = [round(x+offset)-offset for x in self.location]
+			elif self.snapping=="offset":
+				offset=0.5 # could be 0.5
+				ob.location = [round(x+offset)-offset for x in self.location]
+			ob["MCprep_noSwap"] = 1
+
+			if self.make_real:
+				if util.bv28():
+					# make real doesn't select resulting output now (true for
+					# earlier versions of 2.8, not true in 2.82 at least where
+					# selects everything BUT the source of original instance
+					pre_objs = list(bpy.data.objects)
+					bpy.ops.object.duplicates_make_real()
+					post_objs = list(bpy.data.objects)
+					new_objs = list(set(post_objs)-set(pre_objs))
+					for obj in new_objs:
+						util.select_set(obj, True)
+				else:
+					bpy.ops.object.duplicates_make_real()
+
+				if group is not None:
+					self.fix_armature_target(
+						context, context.selected_objects, group)
+
+				# remove the empty added by making duplicates real.
+				if ob in bpy.data.objects[:] and ob.type == "EMPTY":
+					util.obj_unlink_remove(ob, True, context)
+				else:
+					for ob in context.selected_objects:
+						if ob.type != "EMPTY":
+							continue
+						elif ob.parent or ob.children:
+							continue
+						util.obj_unlink_remove(ob, True, context)
+				# TODO: Consider actually keeping the empty, and parenting the
+				# spawned items to this empty (not the defautl behavior)
+			else:
+				# change the draw size
+				if hasattr(ob, "empty_draw_size"):
+					ob.empty_draw_size = 0.25
+				else:
+					ob.empty_display_size = 0.25
+				# mark as active object
+				util.set_active_object(context, ob)
+
+			# Move source of group instance into excluded view layer
+			if util.bv28() and group:
+				move_assets_to_excluded_layer(context, [group])
+
+		self.track_param = self.block
+		return {'FINISHED'}
+
+	def fix_armature_target(self, context, new_objs, src_coll):
```

Comment:
Optional refactor: Move this into `spawn_util.py`, and reference back here for both entities and the meshswap file's equivalent usage.